### PR TITLE
Add full support for analytical windowing functions

### PR DESCRIPTION
**Pull request by jimidle**
Analytical windowing functions were specified in the grammar instead of our standard function lookup table.

These functions are now removed from the parser grammar and specified in the standard tables.

Additionally,  the WITHIN GROUP syntax is supported using an additional `expression` rule alt. This makes function handling orthogonal for all but the so called `builtinFunctions` 

We now support analytical aggregate functions as evinced by the following examples:

```sql
... FIRST_VALUE(Salary) OVER (PARTITION BY DepartmentID ORDER BY Salary DESC)
``` 

```sql
... PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY Salary) OVER (PARTITION BY DepartmentID)
```

Progresses: #357 
Progresses: #275

### CODE REVIEW

File: core/src/test/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionSpec.scala [hunk 1]
```diff
@@ -205,20 +205,60 @@ class TSqlFunctionSpec extends AnyWordSpec with TSqlParserTestCommon with Matche
...
+          ir.CallFunction("PERCENTILE_CONT", Seq(ir.Literal(float = Some(0.5f)))),
+          Seq(ir.SortOrder(ir.Column("Salary"), ir.AscendingSortDirection, ir.SortNullsUnspecified))),
+        Seq(ir.Column("DepartmentID")),
+        List(),
+        ir.WindowFrame(
+          ir.UndefinedFrame,
+          ir.FrameBoundary(current_row = false, unbounded = false, ir.Noop),
+          ir.FrameBoundary(current_row = false, unbounded = false, ir.Noop))))
+//
+//    example(
```
Comment (vil1) [inline core/src/test/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionSpec.scala:250]: Please uncomment/remove this

**Review by vil1 — CHANGES_REQUESTED** [review]
LGTM, minus some commented out code that snuck in.

File: core/src/test/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionSpec.scala (hunk 1 above)
Comment (jimidle) [reply core/src/test/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionSpec.scala]: Too many PRs at once :)

**Comment by jimidle** [comment]
> LGTM, minus some commented out code that snuck in.

@vil1 If you coudl mark the changes as resolved, then @sundarshankar89 can add this to the merge queue - cheers.

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**